Tests/test_SubsMat.py

```diff
@@ -17,65 +17,1433 @@
-with open(ftab_file) as handle:
-    ftab_prot = FreqTable.read_count(handle)
-ctab_file = os.path.join('SubsMat', 'protein_freq.txt')
-with open(ctab_file) as handle:
-    ctab_prot = FreqTable.read_freq(handle)
-f.write("Check differences between derived and true frequencies for each\n")
-f.write("letter. Differences should be very small\n")
-for i in ftab_prot.alphabet.letters:
-    f.write("%s %f\n" % (i, abs(ftab_prot[i] - ctab_prot[i])))
-
-pickle_file = os.path.join('SubsMat', 'acc_rep_mat.pik')
-# Don't want to use text mode on Python 3,
-with open(pickle_file, 'rb') as handle:
-    acc_rep_mat = pickle.load(handle)
-acc_rep_mat = SubsMat.AcceptedReplacementsMatrix(acc_rep_mat)
-obs_freq_mat = SubsMat._build_obs_freq_mat(acc_rep_mat)
-ftab_prot2 = SubsMat._exp_freq_table_from_obs_freq(obs_freq_mat)
-obs_freq_mat.print_mat(f=f, format=" %4.3f")
-
-
-f.write("Diff between supplied and matrix-derived frequencies, should be small\n")
-for i in sorted(ftab_prot):
-    f.write("%s %.2f\n" % (i, abs(ftab_prot[i] - ftab_prot2[i])))
-
-s = 0.
-f.write("Calculating sum of letters for an observed frequency matrix\n")
-counts = obs_freq_mat.sum()
-for key in sorted(counts):
-    f.write("%s\t%.2f\n" % (key, counts[key]))
-    s += counts[key]
-f.write("Total sum %.2f should be 1.0\n" % (s))
-lo_mat_prot = \
-SubsMat.make_log_odds_matrix(acc_rep_mat=acc_rep_mat, round_digit=1)  # ,ftab_prot
-f.write("\nLog odds matrix\n")
-f.write("\nLog odds half matrix\n")
-# Was %.1f. Let us see if this is OK
-lo_mat_prot.print_mat(f=f, format=" %d", alphabet='AVILMCFWYHSTNQKRDEGP')
-f.write("\nLog odds full matrix\n")
-# Was %.1f. Let us see if this is OK
-lo_mat_prot.print_full_mat(f=f, format=" %d", alphabet='AVILMCFWYHSTNQKRDEGP')
-
-f.write("\nTesting MatrixInfo\n")
-for i in MatrixInfo.available_matrices:
-    mat = SubsMat.SeqMat(getattr(MatrixInfo, i))
-    f.write("\n%s\n------------\n" % i)
-    mat.print_mat(f=f)
-f.write("\nTesting Entropy\n")
-relative_entropy = lo_mat_prot.calculate_relative_entropy(obs_freq_mat)
-f.write("relative entropy %.3f\n" % relative_entropy)
-
-f.write("\nmatrix correlations\n")
-blosum90 = SubsMat.SeqMat(MatrixInfo.blosum90)
-blosum30 = SubsMat.SeqMat(MatrixInfo.blosum30)
-f.write("BLOSUM30 & BLOSUM90 %.2f\n" % SubsMat.two_mat_correlation(blosum30, blosum90))
-f.write("BLOSUM90 & BLOSUM30 %.2f\n" % SubsMat.two_mat_correlation(blosum90, blosum30))
+
+class TestGeo(unittest.TestCase):
+
+    ftab_file = os.path.join('SubsMat', 'protein_count.txt')
+    with open(ftab_file) as handle:
+        ftab_prot = FreqTable.read_count(handle)
+    del ftab_file
+
+    pickle_file = os.path.join('SubsMat', 'acc_rep_mat.pik')
+    # Don't want to use text mode on Python 3,
+    with open(pickle_file, 'rb') as handle:
+        acc_rep_mat = pickle.load(handle)
+    acc_rep_mat = SubsMat.AcceptedReplacementsMatrix(acc_rep_mat)
+    del pickle_file
+
+    obs_freq_mat = SubsMat._build_obs_freq_mat(acc_rep_mat)
+
+    del handle
```

Comment:
I'm not sure exactly when this block of parsing code would get executed (in the context of ``run_tests.py`` etc), and thus what would happen if it fails. I think we are better off putting this in a formal class startup method (available in Python 2.7 onwards):

https://docs.python.org/2/library/unittest.html#unittest.TestCase.setUpClass